invenio_app_ils/acquisition/api.py

```diff
@@ -67,6 +77,67 @@ def delete(self, **kwargs):
 order_pid_fetcher = partial(pid_fetcher, provider_cls=OrderIdProvider)
 
 
+class OrderValidator(RecordValidator):
+    """Order record validator."""
+
+    def validate_cancel(self, status, cancel_reason):
+        """Validate decline is correct."""
+        if status == "CANCELLED" and not cancel_reason:
+            raise AcquisitionError(
+                "You have to provide a cancel reason when cancelling the order"
+            )
+        if cancel_reason and not status == "CANCELLED":
+            raise AcquisitionError(
+                "If you select a cancel reason you need to select"
+                " \"Cancelled\" in the state"
+            )
+
+    def ensure_vendor_exists(self, vendor_pid):
+        """Ensure vendor exists or raise."""
+        Vendor = current_ils_acq.vendor_record_cls
+        try:
+            Vendor.get_record_by_pid(vendor_pid)
+        except PIDDoesNotExistError:
+            raise VendorNotFoundError(vendor_pid)
+
+    def ensure_document_exists(self, document_pid):
+        """Ensure document exists or raise."""
+        Document = current_app_ils.document_record_cls
+        try:
+            Document.get_record_by_pid(document_pid)
+        except PIDDoesNotExistError:
+            raise DocumentNotFoundError(document_pid)
+
+    def ensure_patron_exists(self, patron_pid):
```

Comment:
does it fail the validation on the anonymous patron ?

Reply:
Everything will work fine when [this](https://github.com/inveniosoftware/invenio-app-ils/pull/1037/files#diff-d277a329250cd6c47ba4c7764b133e1baca8b6cda3ac234b4e3a142b9cedc468R106) is merged :)